.github/workflows/checks-and-tests.yml

```diff
@@ -23,7 +16,7 @@ jobs:
     - name: Check if code compiles
       run: cargo check --release --all-features --verbose
 
-  test-and-coverage:
+  tests:
```

Comment:
Can we use cache here?

Also, there's the `tarpaulin` below which we use to generate coverage report.
But maybe there's a better approach to this?

I believe I saw some project using UT coverage and the report was published as a comment inside review section.